catalog/provider.go

```diff
@@ -93,35 +140,126 @@ func NewDBProvider(dataDir, dbFile string) (*DatabaseProvider, error) {
 			context.Background(),
 			"CREATE SCHEMA IF NOT EXISTS "+t.Schema,
 		); err != nil {
-			return nil, fmt.Errorf("failed to create internal schema %q: %w", t.Schema, err)
+			return fmt.Errorf("failed to create internal schema %q: %w", t.Schema, err)
 		}
 		if _, err := storage.ExecContext(
 			context.Background(),
 			"CREATE TABLE IF NOT EXISTS "+t.QualifiedName()+"("+t.DDL+")",
 		); err != nil {
-			return nil, fmt.Errorf("failed to create internal table %q: %w", t.Name, err)
+			return fmt.Errorf("failed to create internal table %q: %w", t.Name, err)
 		}
 		for _, row := range t.InitialData {
 			if _, err := storage.ExecContext(
 				context.Background(),
 				t.UpsertStmt(),
 				row...,
 			); err != nil {
-				return nil, fmt.Errorf("failed to insert initial data into internal table %q: %w", t.Name, err)
+				return fmt.Errorf("failed to insert initial data into internal table %q: %w", t.Name, err)
 			}
 		}
 	}
 
-	return &DatabaseProvider{
-		mu:                        &sync.RWMutex{},
-		connector:                 connector,
-		storage:                   storage,
-		catalogName:               name,
-		dataDir:                   dataDir,
-		dbFile:                    dbFile,
-		dsn:                       dsn,
-		externalProcedureRegistry: sql.NewExternalStoredProcedureRegistry(), // This has no effect, just to satisfy the upper layer interface
-	}, nil
+	return nil
+}
+
+func (prov *DatabaseProvider) CreateCatalog(dbFile string) (bool, error) {
```

Comment:
Maybe we should simplify the `*Catalog` APIs to accept a catalog name without the `.db` suffix.